src/rust/engine/process_execution/src/remote.rs

```diff
@@ -663,6 +677,20 @@ impl CommandRunner {
             }
             StreamOutcome::StreamClosed(operation_name_opt) => {
               trace!("wait_on_operation_stream (build_id={}) returned stream close, will retry operation_name={:?}", context.build_id, operation_name_opt);
+
+              // Check if the number of request attempts sent thus far have exceeded the number
+              // of retries allowed since the last successful connection. (There is no point in
+              // continually submitting a request if ultimately futile.)
```

Comment:
I think you could remove this comment - the code is pretty self-documenting.

Reply:
I disagree. I found much the original code here not self-documenting. I'd prefer to leave such notes, especially for my future self as much as anybody else.